src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Cache.cs

```diff
@@ -139,20 +141,18 @@ public static Regex GetOrAdd(string pattern, RegexOptions options, TimeSpan matc
             return regex;
         }
 
-        private static bool TryGet(Key key, [NotNullWhen(true)] out Regex? regex)
+        private static Regex? Get(Key key)
         {
             long lastAccessedStamp = 0;
 
             // We optimize for repeated usage of the same regular expression over and over,
             // by having a fast-path that stores the most recently used instance.  Check
             // to see if that instance is the one we want; if it is, we're done.
-            Node? lastAccessed = s_lastAccessed;
-            if (lastAccessed != null)
+            if (s_lastAccessed is Node lastAccessed)
             {
-                if (lastAccessed.Key.Equals(key))
+                if (key.Equals(lastAccessed.Key))
```

Comment:
For my knowledge - did this line make a perf difference? Or does it just read better this way?

Reply:
It did for an intermediate stage where I was experimenting with passing key as `in`.  I ended up reverting that, but left this as it was as I thought it still read slightly better.